persistence-jdbc/javadsl/src/main/scala/com/lightbend/lagom/javadsl/persistence/jdbc/JdbcPersistenceModule.scala

```diff
@@ -25,6 +26,10 @@ class JdbcPersistenceModule extends AbstractModule {
     bind(classOf[JdbcSession]).to(classOf[JdbcSessionImpl])
     bind(classOf[SlickOffsetStore]).to(classOf[JavadslJdbcOffsetStore])
     bind(classOf[OffsetStore]).to(Key.get(classOf[SlickOffsetStore]))
+    // Play has no explicit binding for DBApiProvider, it uses a just-in-time
+    // binding https://github.com/google/guice/wiki/JustInTimeBindings
+    // We rely on that to substitute our own implementation instead:
+    bind(classOf[DBApiProvider]).to(classOf[LagomDBApiProvider])
```

Comment:
Can we have a test to verify that the Lagom implementation is the one being used? So, if Play latter adds an explicit binding to `DBApiProvider`, we will know that this is not being picked.

Reply:
Is that really needed? Ideally, we should change the behaviour in Play and remove that code in Lagom.

There is a discussion on the Gitter (play contributors). My personal take is that we should not be that strict and let an app to start even if the connection can't be made. Sure there is a value in checking if the connection settings are valid, but failing to start an app because some underlying system is not up is less then ideal.

Reply:
I added the test. It's useful to prove that we are not eagerly connecting to the db. We can remove it later when it becomes obsolete.